packages/mdc-list/foundation.js

```diff
@@ -265,7 +267,11 @@ class MDCListFoundation extends MDCFoundation {
         }
 
         // Explicitly activate links, since we're preventing default on Enter, and Space doesn't activate them.
-        this.adapter_.followHref(currentIndex);
+        if (this.adapter_.getAttributeForElementIndex(currentIndex, 'href')) {
```

Comment:
Do we really need this adapter API? followHref basically already performs this check, and I'm not sure whether these two things should be exclusive (e.g. if the href is not actually causing page navigation, shouldn't it still cause `notifyAction`?)

Reply:
I separated this logic here since browsers trigger synthetic MouseEvent on enter key for anchor elements so that foundation doesn't end up triggering action event twice.

But on a second thought, I removed the need of `followHref` since browsers implicitly trigger `MouseEvent` which is handled by `handleClick` and programmatic `.click()` is unnecessary. For users who want to also handle `Space` key would need to make use of `MDCList:action` custom event. So the new implementation is more aligned to how browsers work with anchor element.

Our drawer is good example to test this. Where <kbd>Enter</kbd> & <kbd>Space</kbd> both triggers `MDCList:action`. Former automatically follows the href, later just triggers the custom event.

Reply:
Interesting...I thought we were calling `preventDefault` on enter, which iirc is half the reason I added `followHref`. I guess if it's valid to assume that lists of links are mutually exclusive with selection/checkbox/radio lists (which _seems_ reasonable), then this is safe.

Reply:
I still have questions/reservations about removing followHref in this PR...

* Does it not seem unnatural for links to process under different circumstances than the action event fires? The other reason `followHref` existed was so that both space and enter would activate links within MDC List, in keeping with e.g. the ARIA menu spec for menu item activation. (One could argue the drawer isn't exactly a menu, but then what is it? We're making the keyboard behave menu-like for it...)
* Does `followHref` _need_ to be removed for this PR? Was it ever incorrectly triggering (or triggering twice? Its removal strikes me as a tangential side effect which is why I've been asking about this.

Reply:
As you mentioned, I think `followHref` was introduced to bring consistent behaviour with <kbd>Enter</kbd> & <kbd>Space</kbd> for anchor elements which I think unnecessary hack.

ARIA navigation menu button [example](https://www.w3.org/TR/wai-aria-practices/examples/menu-button/menu-button-links.html) doesn't even consider <kbd>Space</kbd> key behaviour.

I do not think it was triggering `followHref` twice earlier.